src/tv2-common/helpers/graphics/InternalGraphic.ts

```diff
@@ -52,130 +34,71 @@ export class InternalGraphic {
 		partId?: string,
 		partDefinition?: PartDefinition
 	) {
-		const isStickyIdent = IsStickyIdent(parsedCue)
-
-		const engine = parsedCue.target
-
 		const mappedTemplate = GetFullGraphicTemplateNameFromCue(config, parsedCue)
 
-		const sourceLayerId = GetSourceLayerForGraphic(config, mappedTemplate, isStickyIdent)
+		const sourceLayerId = GetSourceLayerForGraphic(config, mappedTemplate)
 
 		this.config = config
 		this.parsedCue = parsedCue
 		this.partDefinition = partDefinition
 		this.adlib = adlib
 		this.mappedTemplate = mappedTemplate
-		this.isStickyIdent = isStickyIdent
 		this.engine = parsedCue.target
 		this.name = GraphicDisplayName(config, parsedCue)
 		this.sourceLayerId = sourceLayerId
-		this.outputLayerId = IsTargetingWall(engine) ? SharedOutputLayers.SEC : SharedOutputLayers.OVERLAY
+		this.outputLayerId = IsTargetingWall(this.engine) ? SharedOutputLayers.SEC : SharedOutputLayers.OVERLAY
 		this.partId = partId
 		this.content = this.getInternalGraphicContent()
 	}
 
-	public createAdlibTargetingOVL(
-		context: ICommonContext,
-		actions: IBlueprintActionManifest[],
-		adlibPieces: IBlueprintAdLibPiece[]
-	): void {
-		if (IsTargetingOVL(this.engine) && this.isStickyIdent) {
-			const userData = literal<ActionPlayGraphics>({
-				type: AdlibActionType.PLAY_GRAPHICS,
-				graphic: this.parsedCue
-			})
-			actions.push(
-				literal<IBlueprintActionManifest>({
-					externalId: generateExternalId(context, userData),
-					actionId: AdlibActionType.PLAY_GRAPHICS,
-					userData,
-					userDataManifest: {},
-					display: {
-						_rank: this.rank || 0,
-						label: t(this.name),
-						uniquenessId: `gfx_${this.name}_${this.sourceLayerId}_${this.outputLayerId}_commentator`,
-						sourceLayerId: this.sourceLayerId,
-						outputLayerId: SharedOutputLayers.OVERLAY,
-						tags: [AdlibTags.ADLIB_KOMMENTATOR],
-						content: _.clone(this.content)
-					}
-				})
-			)
-		} else if (IsTargetingOVL(this.engine)) {
-			const adLibPiece = literal<IBlueprintAdLibPiece>({
+	public createCommentatorAdlib(adlibPieces: IBlueprintAdLibPiece[]): void {

```

Comment:
Only have one comment for this method and that is that it produces an side effect.
It modifies an external array.  

Ideally, instead of taking the `adlibPieces` array as a parameter and modifying it, it should just return the newly created commentator adlib and then the caller of this method has the responsibility of adding it to an array (or whatever the commentator adlib is needed for. Might not be in an array in some places)

It also makes more sense (atleast for me) that functions named `create` are not void functions, but instead returns the things they are creating.

However, the practice of giving an array as a parameter and then modifying it is quite common in Sofie and a lot of logic is dependent on this quirk. So I will let you decide if you refactor it now or not. (But I most definately will comment on it again in other PRs ;) )